models/clan.go

```diff
@@ -978,10 +978,18 @@ func SearchClan(
 		return clans, nil
 	}
 
+	var filter interface{}
+	if useTextSearch {
+		filter = bson.M{"$text": bson.M{"$search": term}}
+	} else {
+		escapedTerm := fmt.Sprintf(`^\Q%s\E`, term)
+		filter = bson.M{"name": bson.M{"$regex": escapedTerm}}
```

Comment:
Golang Regex engine is guaranteed to run at linear time based on the length of the input.
https://golang.org/pkg/regexp/
```
The regexp implementation provided by this package is guaranteed to run in time linear in the size of the input. (This is a property not guaranteed by most open source implementations of regular expressions.)
```
Do you know if Mongo guarantees the same?

Reply:
The implication if it does not follow is that the user can pass some exponential time regexes: https://stackoverflow.com/questions/8887724/why-can-regular-expressions-have-an-exponential-running-time

Reply:
Mongo uses PCRE, which does not have such guarantees. However, the regex /^\Q...\E/ is a prefix expression with no operators, and according to this https://docs.mongodb.com/manual/reference/operator/query/regex/#index-use,
> /^a/ can stop scanning after matching the prefix
so it should be pretty efficient.
I would write the load test to confirm, but I'm not sure I understood how to do that.